arbnode/node.go

```diff
@@ -660,7 +660,7 @@ func createNodeImpl(
 			confirmedNotifiers = append(confirmedNotifiers, messagePruner)
 		}
 
-		stakerObj, err = staker.NewStaker(l1Reader, wallet, bind.CallOpts{}, config.Staker, blockValidator, statelessBlockValidator, nil, confirmedNotifiers, deployInfo.ValidatorUtils, fatalErrChan)
+		stakerObj, err = staker.NewStaker(l1Reader, wallet, bind.CallOpts{}, func() *staker.L1ValidatorConfig { return &configFetcher.Get().Staker }, blockValidator, statelessBlockValidator, nil, confirmedNotifiers, deployInfo.ValidatorUtils, fatalErrChan)
```

Comment:
This means that the options in `Staker` will only be loaded during startup, the hot reloadable flag will be ignored.

Reply:
Since the staker now takes a function of type `L1ValidatorConfigFetcher` which is a function that would call configFetcher every time we invoke it this effectively implements live reloading. 
(this impl is similar to how its done for other hot-reloadable confis such as batchposterconfig https://github.com/OffchainLabs/nitro/blob/7ccb6ad0ea23863c0c6e4fee39026ca9b0d573fc/arbnode/node.go#L697)

I've tested this locally that hot-reloadable options are updated periodically.

Reply:
:+1: